io/GDALReader.cpp

```diff
@@ -81,6 +83,8 @@ void GDALReader::initialize()
     m_count = m_raster->width() * m_raster->height();
     m_bandTypes = m_raster->getPDALDimensionTypes();
     m_raster->close();
+    m_row = 0;
```

Comment:
These belong in ready().  That allows execute() to be called more than once.  Yes, this probably never happens.

Reply:
The rules of when to do which aren't so clear.